src/ServiceControl.Monitoring/Metrics/Raw/LongValueOccurenceSerializerDefinition.cs

```diff
@@ -24,33 +24,52 @@ public void Serialize(object message, Stream stream)
 
         public object[] Deserialize(Stream stream, IList<Type> messageTypes = null)
         {
-            var message = new LongValueOccurrences();
+            var reader = new BinaryReader(stream);
 
-            ToLongValueOccurrences(message, stream);
+            var version = reader.ReadInt64();
 
-            return new object[]
+            if (version == 1)
             {
-                message
-            };
-        }
+                var baseTicks = reader.ReadInt64();
+                var count = reader.ReadInt32();
 
-        static void ToLongValueOccurrences(LongValueOccurrences message, Stream stream)
-        {
-            var reader = new BinaryReader(stream);
+                if (count == 0)
+                {
+                    // no records, return empty
+                    return new object[0];
+                }
 
-            message.Version = reader.ReadInt64();
-            message.BaseTicks = reader.ReadInt64();
+                var messages = new List<object>(1); // usual case
+                var message = LeaseMessage(messages);
 
-            var count = reader.ReadInt32();
+                for (var i = 0; i < count; i++)
+                {
+                    var ticks = reader.ReadInt32();
+                    var value = reader.ReadInt64();
+                    var date = baseTicks + ticks;
 
-            message.Ticks = new int[count];
-            message.Values = new long[count];
+                    if (message.TryRecord(date, value) == false)
+                    {
+                        message = LeaseMessage(messages);
```

Comment:
@danielmarbach I think a behavior in the transport receive context would do the trick. I talked about this earlier with @tmasternak but we agreed that serializer was good enough at that stage. As I see, you are trying to squeeze tiny bits of performance in the PRs now. I suggest you drop the serializer and the whole pipeline and use the transport receive context. Or the `NServiceBus.Raw` :-). It is now used in Wormhole gateway, Bridge and the SC.TransportAdapter so you won't be my first customers ;-)